src/carnot/funcs/builtins/regex_ops.h

```diff
@@ -110,6 +112,55 @@ class RegexReplaceUDF : public udf::ScalarUDF {
   std::unique_ptr<re2::RE2> regex_;
 };
 
+class MatchRegexRule : public udf::ScalarUDF {
+ public:
+  Status Init(FunctionContext* ctx, StringValue encodedRegexRules) {
+   // parse encodedRegexRules as json
+   rapidjson::Document regex_rules;
+   rapidjson::ParseResult parse_result = regex_rules.Parse(encodedRegexRules.data());
+   if (!parse_result) {
+     return Status(statuspb::Code::INVALID_ARGUMENT, "unable to parse string as json");
+   }
+   // populate self::regex_rules_map_
+   for (rapidjson::Value::ConstMemberIterator itr = regex_rules.MemberBegin(); 
+        itr != regex_rules.MemberEnd(); ++itr) {
+    RegexMatchUDF regex_match_udf; 
+    auto name = itr->name.GetString();
+    auto regex_pattern = itr->value.GetString();
+    PL_RETURN_IF_ERROR(regex_match_udf.Init(ctx, regex_pattern)); 
+    regex_rules_map_.emplace(name, std::move(regex_match_udf));
+   }
+   return Status::OK();
+  }
+  
+  types::StringValue Exec(FunctionContext *ctx, StringValue value) {
+    for (auto& [rule_name, regex_match_udf] : regex_rules_map_) {
+      if (regex_match_udf.Exec(ctx, value).val) {
+        return rule_name;
+      }
+    }
+  return "";
```

Comment:
I've changed the internal data structure from a map to a vector pair. I also added a length variable to keep track of the size/length of the vector since I was having trouble using .size() to automatically determine it (this was returning a long instead of an int for comparison). Let me know if there's a better way to do this!